Applications/Spire/Source/Ui/QuantityBox.cpp

```diff
@@ -34,4 +35,50 @@ QuantityBox::QuantityBox(std::shared_ptr<OptionalQuantityModel> model,
   update_style(get_decimal_box(), [&] (auto& style) {
     style.get(Any()).set(TrailingZeros(0));
   });
+  get_decimal_box().installEventFilter(this);
+}
+
+bool QuantityBox::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == &get_decimal_box() && event->type() == QEvent::Show) {
+    auto proxy = find_focus_proxy(get_decimal_box());
+    if(proxy) {
+      proxy->installEventFilter(this);
+    }
+  } else if(event->type() == QEvent::KeyPress && !is_read_only()) {
+    auto& key_event = *static_cast<QKeyEvent*>(event);
+    auto multiplier = 1.0;
+    if(key_event.key() == Qt::Key_K) {
+      multiplier = 1000;
+    } else if(key_event.key() == Qt::Key_H) {
+      multiplier = 100;
+    } else if(key_event.key() == Qt::Key_D) {
+      if(key_event.modifiers().testFlag(Qt::AltModifier)) {
+        multiplier = 10;
+      } else {
+        multiplier = 0.1;
```

Comment:
Work with integers when possible. Instead of ```0.1 * x```, rewrite it as ```x / 10```. This avoids introducing any additional floating point issues.